source/main.cpp

```diff
@@ -1,65 +1,54 @@
-/*
-The MIT License (MIT)
-
-Copyright (c) 2016 British Broadcasting Corporation.
-This software is provided by Lancaster University by arrangement with the BBC.
-
-Permission is hereby granted, free of charge, to any person obtaining a
-copy of this software and associated documentation files (the "Software"),
-to deal in the Software without restriction, including without limitation
-the rights to use, copy, modify, merge, publish, distribute, sublicense,
-and/or sell copies of the Software, and to permit persons to whom the
-Software is furnished to do so, subject to the following conditions:
-
-The above copyright notice and this permission notice shall be included in
-all copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
-THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-DEALINGS IN THE SOFTWARE.
-*/
-
-// Works with the Animal Vegetable Mineral guessing game in the Android 'micro:bit Blue' app which is obtainable from
-// https://play.google.com/store/apps/details?id=com.bluetooth.mwoolley.microbitbledemo
-
 #include "MicroBit.h"
 #include "MicroBitUARTService.h"
 
 MicroBit uBit;
 MicroBitUARTService *uart;
 
+const MicroBitImage IMAGE_HAPPY = MicroBitImage("0,0,0,0,0\n0,1,0,1,0\n0,0,0,0,0\n1,0,0,0,1\n0,1,1,1,0\n");  
+
 int connected = 0;
 
 void onConnected(MicroBitEvent)
 {
-
-    uBit.display.scroll("C");
-
+    uBit.display.print(IMAGE_HAPPY);
     connected = 1;
+    uart->eventOn(ManagedString("\n"), ASYNC);
+}
 
-    // mobile app will send ASCII strings terminated with the colon character
-    ManagedString eom(":");
-
-    while(connected == 1) {
-        ManagedString msg = uart->readUntil(eom);
-        uBit.display.scroll(msg);
+void onUartEvent(MicroBitEvent)
+{
+    ManagedString msg = uart->readUntil(ManagedString(":"), ASYNC);
+    if (msg.length() == 0)
+    {
+        // Invalid command. Throw out rx buffer.
+        uart->readUntil(ManagedString("\n"));
+        return;
     }
 
+    if (msg == "disp")

```

Comment:
Even though this is only one for now, it will grow to handle many commands. It might make sense to make this a `switch` statement and to put the contents of each into separate functions. You could also use a [map](http://www.cplusplus.com/reference/map/map/) with `disp` as the key and a pointer to the handler function as the value to remove the switch completely.